incidents_store.go

```diff
@@ -0,0 +1,131 @@
+package main
+
+import (
+	"DocPlanner/pingdom-statuspage-integration/statuspage"
+	"fmt"
+	"sync"
+	"time"
+)
+
+const DEFAULT_INCIDENT_LAG_SECONDS = 60
+
+type incidentClient struct {
+	StatusPageClient *statuspage.Client
+}
+
+type incidentStore struct {
+	components []*component
+	mux        sync.Mutex
+}
+
+type component struct {
+	timestamp int64
+	id        string
+	pageId    string
+}
+
+func NewIncidentClient(statusPageClient *statuspage.Client) *incidentClient {
+	return &incidentClient{
+		StatusPageClient: statusPageClient,
+	}
+}
+
+func (is *incidentStore) Add(componentID string, pageID string, timestamp int64) {
+	is.mux.Lock()
+	candidate := &component{
+		id:        componentID,
+		pageId:    pageID,
+		timestamp: timestamp,
+	}
+	is.components = append(is.components, candidate)
+	is.mux.Unlock()
+}
+
+func (is *incidentStore) GetAll() []*component {
+	is.mux.Lock()
+	defer is.mux.Unlock()
+	return is.components
+}
+
+func (is *incidentStore) Remove(componentID string) {
+	is.mux.Lock()
+	componentIndex := 0
+	for index, component := range is.components {
+		if component.id == componentID {
+			componentIndex = index
+			break
+		}
+	}
+	if len(is.components) > 0 {
+		is.components = append(is.components[:componentIndex], is.components[componentIndex+1:]...)
+	}
+	is.mux.Unlock()
+}
+
+func (is *incidentStore) CheckEvaluation() []*component {
+
+	var components []*component
+	incidentStore := is.GetAll()
+	now := time.Now().Unix()
+
+	for _, candidate := range incidentStore {
+		relative := now - candidate.timestamp
+		if relative > DEFAULT_INCIDENT_LAG_SECONDS {
```

Comment:
It'll be good idea to use time.Before/time.After functions because the are based on monotonic time.